modules/swift/pages/reactive.adoc

```diff
@@ -0,0 +1,506 @@
+    var title: String
+    var completed: Bool
+}
+----
+====
+
+== Manipulate Documents as Codable Model Objects
+
+Once defined, document models can be persisted, retrieved, and removed using type-safe APIs on the Collection object.
+
+=== Get a Document as a Decodable Object
+
+To get a document from a collection and decode it into a Swift model:
+
+* Use the `document` function with the collection.
+* Specify the document ID and type.
+
+====
+[source,swift]
+----
+----
+====
+
+// FIXME: API link
+For API details, see {url-api-references}/Classes/Collection.html#/Document%20Management[Document Management].
+
+=== Save an Encodable Object as a Document
+
+To save a document represented by a Swift model object into a collection:
+
+* Use the `save` function with the collection.
+* Specify the object.
+* Optionally, specify a concurrency control or a conflict handler.
+By default, last-write-wins concurrency control will be used if a conflict occurs.
+
+====
+[source,swift]
+----
+----
+====
+
+// FIXME: API link
+For API details, see {url-api-references}/Classes/Collection.html#/Document%20Management[Document Management].
+
+=== Delete a Document
+
+To remove a document represented by a Swift model object from a collection:
+
+* Use the `delete` function with the collection.
+* Specify the object.
+* Optionally, specify a concurrency control.
+By default, last-write-wins concurrency control will be used if a conflict occurs.
+
+====
+[source,swift]
+----
+----
+====
+
+// FIXME: API link
+For API details, see {url-api-references}/Classes/Collection.html#/Document%20Management[Document Management].
+
+=== Purge a Document
+
+Purging a document is permanent deletion, not replicated.
+
+To purge a document represented by a Swift model object from a collection:
+
+* Use the `purge` function with the collection.
+* Specify the object.
+
+====
+[source,swift]
+----
+----
+====
+
+// FIXME: API link
+For API details, see {url-api-references}/Classes/Collection.html#/Document%20Management[Document Management].
+
+== Decode Query Results
+
+The Reactive APIs enable you to decode Couchbase {sqlpp} query results directly into model objects.
+This makes queries observable, so that you can refresh the application automatically as the results change.
+
+With prior solutions such as xref:swift:query-live.adoc[Live Queries], you need to trigger user interface updates manually inside the listener.
+This can lead to complex and difficult-to-maintain state management.
+With the Reactive APIs, you can use SwiftUI's minimalistic and declarative approach to manage querying and updating the user interface.
+
+To make query results observable, the model object only needs to conform to the decodable protocol, and does not require a document ID property annotated with the `@DocumentID` property wrapper.
+However, if you want to save the model to a collection later, it must include a document ID property with the `@DocumentID` property wrapper.
```

Comment:
can we please have an empty line before this?

and maybe reword to something like `However, if you wish to save any result later, the decoding model must include a document ID property with the `@DocumentID` property wrapper."